src/source/tile_id.js

```diff
@@ -164,11 +164,26 @@ export class OverscaledTileID {
     }
 }
 
-function calculateKey(wrap: number, z: number, x: number, y: number) {
+function calculateKey(wrap: number, overscaledZ: number, z: number, x: number, y: number): string {
     wrap *= 2;
     if (wrap < 0) wrap = wrap * -1 - 1;
     const dim = 1 << z;
-    return ((dim * dim * wrap + dim * y + x) * 32) + z;
+    let id = `${dim * dim * wrap + dim * y + x}`;
+    // Always reserve two characters for both z values in order to
+    // keep the sorting behavior consistent
+    if (z < 10)
+        id += "0";
+    id += z;
+    if (overscaledZ < 10)
+        id += "0";
+    id += overscaledZ;
+    return id;
+}
+
+export function tileIDKeyComparison(keyA: string, keyB: string): number {
```

Comment:
I'm not sure if we need this functionality at all. The only place where tiles are being sorted based on their zoom-level is `getIds()` and 'getRenderableIds()` in source_cache, but I've already modified that code to use tile ids instead of keys. Do we have any other requirements for the order of keys that I might have missed?

Reply:
Let's remove it if it's not used :)